ai-assistant/src/ContentRenderer.tsx

```diff
@@ -215,16 +215,43 @@ const ContentRenderer: React.FC<ContentRendererProps> = React.memo(
           (className === 'language-json' || isJsonKubernetesResource(children));
 
         if (isYamlBlock && onYamlDetected && typeof children === 'string') {
-          // Try to parse as Kubernetes YAML
-          const parsed = parseKubernetesYAML(children);
-          if (parsed.isValid) {
+          // Handle multi-document YAML (documents separated by ---)
+          const yamlDocuments = children.split(/^---+$/m).filter(doc => doc.trim());
```

Comment:
Why do you split the documents? If we have them in one doc maybe it's easier to apply? Or is this because the apply dialog you use has a limitation there?